addons/mail/static/src/core/messaging_service.js

```diff
@@ -191,20 +191,28 @@ export class Messaging {
                     break;
                 case "discuss.channel/leave":
                     {
-                        const thread = this.threadService.insert({
-                            ...notif.payload,
-                            model: "discuss.channel",
-                        });
-                        this.threadService.remove(thread);
-                        if (thread.localId === this.store.discuss.threadLocalId) {
-                            this.store.discuss.threadLocalId = undefined;
-                        }
+                        const thread = this._removeThread(notif.payload);
                         this.notificationService.add(
                             sprintf(_t("You unsubscribed from %s."), thread.displayName),
                             { type: "info" }
                         );
                     }
                     break;
+                case "discuss.channel/delete":
+                    {
+                        const thread = this._removeThread({ id: notif.payload.id });
+                        this.store.discuss.starred.messages = this.store.discuss.starred.messages.filter(
+                            (message) => message.resId !== thread.id && message.resModel !== thread.model
+                        );
+                        this.store.discuss.inbox.messages = this.store.discuss.inbox.messages.filter(
+                            (message) => message.resId !== thread.id && message.resModel !== thread.model
+                        );
+                        this.store.discuss.inbox.counter -= thread.message_needaction_counter;
+                        thread.messages.forEach((message) => {
+                            delete this.store.messages[message.id];
+                        });
+                    }
+                    break;
```

Comment:
Please configure `web/tooling` in your IDE, so that it follows prettier/eslint configs.
Otherwise any changes in this PR will require diffing more than necessary. And also formatting helps reviewing code more efficiently.